aws/cloudformation/bootstrap_frontend.sh.erb

```diff
@@ -20,7 +20,10 @@ ps aux | grep "cluster worker" | grep dashboard | head -n 3 | tr --squeeze-repea
 export AWS_METADATA_SERVICE_TIMEOUT=30
 export AWS_METADATA_SERVICE_NUM_ATTEMPTS=30
 
-# Make sure server passes health check.
+# Ensure the instance launch has completed successfully by requesting the Health Check endpoint for Pegasus via nginx
+# (localhost:80/health_check) and also Dashboard directly via its Unix socket, bypassing nginx.
 # The status code determines success/failure of the instance launch.
 curl --silent --show-error --fail --retry 10 --retry-connrefused --retry-max-time 300 \
-  localhost:80/health_check > /dev/null
+  localhost:80/health_check > /dev/null \
+&& curl --silent --show-error --fail --retry 10 --retry-connrefused --retry-max-time 300 \
+  --unix-socket /run/unicorn/dashboard.sock localhost:80/health_check > /dev/null
```

Comment:
I don't love hardcoding this value, or baking specific knowledge of the details of our socket files into this bootstrap script. Is there a way we could implement this more generically?

Reply:
Agreed. Sorry I didn't explain why I took this shortcut. The original check sends an http request to `http://localhost:80/health_check`. `nginx` listens on port 80 on each web application server and uses the `host` header to determine [whether to route the request](https://github.com/code-dot-org/code-dot-org/blob/4764dd4f32c7bb6a06504a3a9b0b15b720ebcd86/cookbooks/cdo-nginx/templates/default/nginx.conf.erb#L34-L42), falling back to Pegasus if there's no match for the host name. So the original check was testing whether it was possible to successfully invoke the Pegasus health check endpoint via nginx.

I've updated the curl requests to explicitly set the host header (`-H "Host: studio"`). Dynamically populating this shell script with the fully qualified domain name is a bit complex. Since nginx is only checking for the string `studio` in the hostname, we could use this less hacky option